### ISSUE TITLE
Add HostSNIRegexp

### ISSUE THREAD

**Issue by varac** [traefik/traefik#7266]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

I'd like to have a `HostSNIRegexp` so that I can route all subdomains to a certain service, without referencing them all at once.
I.e. similar to the `HostRegexp` rule:

    HostSNIRegexp(`example.com`, `{subdomain:[a-z]+}.example.com`, ...)

That would save time with static configs and a lot of subdomains.

**Comment by ldez** [comment]
related to  #6003

**Comment by FuNK3Y** [comment]
Any news about this one? On the related PR it was mentioned that a discussion should take place about the replacement of mux. Is it the right place?

**Comment by varac** [comment]
I also don't understand if #6003 fixes/closes this.. ?

**Comment by varac** [comment]
Ok, it seems this is still not implemented. #6003 was closed and *not* merged.

**Comment by nmasse-itix** [comment]
Hi ! 

I'm interested in this feature (using it and eventually contributing to). 

How can we make this issue progress one step further ?
Has the new regex syntax mentioned in #7635 been implemented ?

> As explained in # 6003, we think that the mux syntax for regex is not perfect. For a new regex matcher we think it's not a good idea to use the mux syntax, but if we introduce a new syntax, we need to handle it in both TCP and HTTP regex matches.

**Comment by Blackclaws** [comment]
This is also relevant to me. I have a downstream cluster that handles all requests for a certain subdomain and I cannot know in advance what the domains will be. TLS Termination in traefik is not an option unfortunately.

**Comment by JeanTekk** [comment]
I would be also glad about such a feature.
I am have some docker services on a different network. The docker host runs also a traefik instance. The traefik on the docker host routes all containers on common ports and does the detection of the services and ssl via labels. 
The docker host forwads the traefik ports for web and websecure(ssl) over an ssh tunnel to another publicly accessible  gatway.
From there i would like to route all requests matching (e.g. "*.docker.myhost.com") to the tunnels.

**Comment by Soffid** [comment]
Hello all.

I've faced the same problem, and managed to patch Traefik 2.5.4 to allow wildcards in TCP  rules. Now, I can write a rule like this one:

- match: HostSNI(`iam3-gw.soffid.com`)||HostSNI(`*.iam3-gw.soffid.com`) 
 
This is the first time I code in GO and I am not a Traefik expert, but as long as we have tested, it has worked fine.

Hope it helps you and can be merged in future versions.

[0001-Allow-wildcards-in-HostSNI.patch.txt](https://github.com/traefik/traefik/files/7679002/0001-Allow-wildcards-in-HostSNI.patch.txt)
.
